app/models.py

```diff
@@ -722,7 +733,7 @@ class ServiceTableMixin(object):
     # Service publishing time.
     service_id = db.Column(db.String, index=True, unique=True, nullable=False)
 
-    data = db.Column(JSON)
+    data = db.Column(JSON, nullable=False, default=dict)
```

Comment:
For services I'm pretty sure they are, since all endpoints should try to assign something to `data`. Tests I'm not sure about, so if there are a lot failing without this then maybe it's not a big deal to keep the default. Up to you.